### PR TITLE
Serialization removal

### PR DESCRIPTION
**Pull request by MiYanni**
Removed order information from code model, as it was used for serialization. Minor code cleanup and refactoring.

### CODE REVIEW

File: src/AutoRest.CSharp.V3/Plugins/Modeler.cs [hunk 1]
```diff
@@ -201,40 +172,22 @@ private PathSegment[] ToPathParts(string httpRequestUri, Dictionary<string, Cons
...
-                    };
-                case ObjectSchema objectSchema:
-                    return new ClientObject(objectSchema, objectSchema.CSharpName(),
-                        objectSchema.Properties.Where(property => !(property.Schema is ConstantSchema)).Select(CreateProperty),
-                        objectSchema.Properties.Where(property => property.Schema is ConstantSchema).Select(CreateConstant));
-            }
-
-            throw new NotImplementedException();
-        }
+            SealedChoiceSchema sealedChoiceSchema => (ClientModel.ClientModel)new ClientEnum(sealedChoiceSchema,
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Plugins/Modeler.cs]: This didn't look great but now I feel it's even worse.

File: test/AutoRest.TestServer.Tests/body-complex/Generated/Models/Sawshark.Serialization.cs [hunk 2]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
```
Comment (pakrym) [inline test/AutoRest.TestServer.Tests/body-complex/Generated/Models/Sawshark.Serialization.cs]: Which generator change fixed this?

File: src/AutoRest.CSharp.V3/Pipeline/Extensions.cs [hunk 3]
```diff
@@ -85,102 +83,142 @@ internal static class Extensions
...
+            writer.Append(name);
+            writer.Append(isNullable ? "?" : string.Empty);
+            writer.Line(".Serialize(writer);");
+        }
+
+        private static void WriteSerializeClientEnum(WriterBase writer, string name, bool isNullable, bool isStringBased)
+        {
+            writer.Append("writer.WriteStringValue(");
+            writer.Append(name);
+            writer.Append((!isStringBased && isNullable) ? ".Value" : string.Empty);
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Pipeline/Extensions.cs]: I don't like this pattern. If the alternative is not to do anything why not use if?

File: src/AutoRest.CSharp.V3/Plugins/Modeler.cs (hunk 1 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/Plugins/Modeler.cs]: Addressed. Put these into separate creation methods.

File: test/AutoRest.TestServer.Tests/body-complex/Generated/Models/Sawshark.Serialization.cs (hunk 2 above)
Comment (MiYanni) [reply test/AutoRest.TestServer.Tests/body-complex/Generated/Models/Sawshark.Serialization.cs]: It wasn't a generator change. Unfortunately, we commit generated code (and don't mark it generated yet (which we have an issue to do)). However, I told resharper to clean up my unused usings, and it hit the generated files. I can revert.

File: src/AutoRest.CSharp.V3/Pipeline/Extensions.cs (hunk 3 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/Pipeline/Extensions.cs]: It is data driven design over logic driven design. I'll change this to be an if-statement.

File: src/AutoRest.CSharp.V3/ClientModel/ClientEnum.cs [hunk 4]
```diff
@@ -8,11 +8,12 @@ namespace AutoRest.CSharp.V3.ClientModel
 {
     internal class ClientEnum : ClientModel, ISchemaTypeProvider
     {
-        public ClientEnum(Schema schema, string name, IEnumerable<ClientEnumValue> values)
+        public ClientEnum(Schema schema, string name, IEnumerable<ClientEnumValue> values, bool isStringBased = false)
         {
             Schema = schema;
             Name = name;
             Values = new List<ClientEnumValue>(values);
+            IsStringBased = isStringBased;
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/ClientModel/ClientEnum.cs]: Remove the setter.

File: src/AutoRest.CSharp.V3/Plugins/Modeler.cs [hunk 5]
```diff
@@ -243,30 +207,22 @@ private static ClientObjectConstant CreateConstant(Property property)
...
-            }
-        }
-
+            BinarySchema _ => (ClientTypeReference)new BinaryTypeReference(false),
+            ByteArraySchema _ => new BinaryTypeReference(false),
+            { Type: AllSchemaTypes.Binary } => new BinaryTypeReference(false),
+            ArraySchema array => new CollectionTypeReference(CreateType(array.ElementType, false)),
+            DictionarySchema dictionary => new DictionaryTypeReference(new FrameworkTypeReference(typeof(string)), CreateType(dictionary.ElementType, isNullable)),
+            //https://devblogs.microsoft.com/dotnet/do-more-with-patterns-in-c-8-0/
+            { } s when s.Type.ToFrameworkCSharpType() is { } type => new FrameworkTypeReference(type, isNullable),
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Plugins/Modeler.cs]: I know resharper suggests this but do you think it's more readable than `Schema s when s.Type.ToFrameworkCSharpType() is Type`?

File: src/AutoRest.CSharp.V3/ClientModel/ClientEnum.cs (hunk 4 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/ClientModel/ClientEnum.cs]: Done.

File: src/AutoRest.CSharp.V3/Plugins/Modeler.cs (hunk 5 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/Plugins/Modeler.cs]: Fixed.

**Pull request merged**